modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -652,8 +677,29 @@ export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
       throw new Error('Missing BitGo GPG key for MPCv2');
     }
 
-    const { derivationPath, hashBuffer } = this.getHashStringAndDerivationPath(txRequest, requestType);
+    if (requestType === RequestType.tx) {
+      assert(txRequest.transactions || txRequest.unsignedTxs, 'Unable to find transactions in txRequest');
+      const unsignedTx =
+        txRequest.apiVersion === 'full' ? txRequest.transactions![0].unsignedTx : txRequest.unsignedTxs[0];
+      txOrMessageToSign = unsignedTx.signableHex;
+      derivationPath = unsignedTx.derivationPath;
+      bufferContent = Buffer.from(txOrMessageToSign, 'hex');
+    } else if (requestType === RequestType.message) {
+      txOrMessageToSign = txRequest.messages![0].messageEncoded;
+      derivationPath = txRequest.messages![0].derivationPath || 'm/0';
+      bufferContent = Buffer.from(txOrMessageToSign);
+    } else {
+      throw new Error('Invalid request type');
+    }
 
+    let hash: Hash;
+    try {
+      hash = this.baseCoin.getHashFunction();
+    } catch (err) {
+      hash = createKeccakHash('keccak256') as Hash;
+    }
+    // check what the encoding is supposed to be for message
+    const hashBuffer = hash.update(bufferContent).digest();
```

Comment:
i would prefer if you modify getHashStringAndDerivationPath to work with messages, but i can do later when external signer is implemented for messages.